src/mozilla_taskgraph/transforms/replicate.py

```diff
@@ -0,0 +1,248 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from __future__ import annotations
+
+import os
+import re
+from textwrap import dedent
+
+from taskgraph.transforms.base import TransformSequence
+from taskgraph.util.attributes import attrmatch
+from taskgraph.util.schema import Schema
+from taskgraph.util.taskcluster import (
+    find_task_id,
+    get_artifact,
+    get_task_definition,
+)
+from voluptuous import ALLOW_EXTRA, Any, Optional, Required
+
+REPLICATE_SCHEMA = Schema(
+    {
+        Required(
+            "replicate",
+            description=dedent(
+                """
+            Configuration for the replicate transforms.
+            """.lstrip(),
+            ),
+        ): {
+            Required(
+                "target",
+                description=dedent(
+                    """
+                Define which tasks to target for replication.
+
+                Each item in the list can be either:
+
+                    1. A taskId
+                    2. An index path that points to a single task
+
+                If any of the resolved tasks are a Decision task, targeted
+                tasks will be derived from the `task-graph.json` artifact.
+                """.lstrip()
+                ),
+            ): [str],
+            Optional(
+                "include-attrs",
+                description=dedent(
+                    """
+                A dict of attribute key/value pairs that targeted tasks will be
+                filtered on. Targeted tasks must *match all* of the given
+                attributes or they will be ignored.
+
+                Matching is performed by the :func:`~taskgraph.util.attrmatch`
+                utility function.
+                """.lstrip(),
+                ),
+            ): {str: Any(str, [str])},
+            Optional(
+                "exclude-attrs",
+                description=dedent(
+                    """
+                A dict of attribute key/value pairs that targeted tasks will be
+                filtered on. Targeted tasks must *not match any* of the given
+                attributes or they will be ignored.
+
+                Matching is performed by the :func:`~taskgraph.util.attrmatch`
+                utility function.
+                """.lstrip(),
+                ),
+            ): {str: Any(str, [str])},
+            Optional(
+                "include-deps",
```

Comment:
This is not implemented :). Presumably you were foregoing this for now?

Reply:
Oops, good catch!

I think this feature only makes sense if we're running integration tests across TC instances? The ability to run across TC instances is hyper specific to fxci-config (I think), so my plan was for mozilla-taskgraph to not support that in any way (and for fxci-config to layer that ability on top of these transforms)

Reply:
Yeah, I think it's fine not to support this here.